app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+        // @todo research a more official way to replace .env values... I could not find one
+        $contents = file_get_contents($this->projectpath.DIRECTORY_SEPARATOR.'.env');
+
+        $newcontents = $contents;
+        preg_match("@$name=(.*)@", $contents, $matches);
+        if (isset($matches[1])) {
+            // @todo sanitize new value and research .env guidelines
+            $newcontents = str_replace("$name=$matches[1]", "$name=$newvalue", $newcontents);
+        }
+
+        file_put_contents($this->projectpath.DIRECTORY_SEPARATOR.'.env', $newcontents);
+
+        return ($newcontents !== $contents) ? true : false;
+    }
+
+    protected function doNodeOrYarn()
+    {
+        $command = '';
+        if ($this->tools['yarn']) {
+            $command = 'yarn';
+        } elseif ($this->tools['npm']) {
+            $command = 'npm install';
+        }
+
+        if (empty($command)) {
+            $this->error('Either yarn or npm are required');
+
+            return false;
+        }
+
+        $this->info("Executing $command now; in {$this->projectpath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+        $process->start();
+
+        $iterator = $process->getIterator($process::ITER_SKIP_ERR | $process::ITER_KEEP_OUTPUT);
+        foreach ($iterator as $data) {
+            $this->line($data);
+        }
+    }
+
+    protected function doAuth()
+    {
+        $command = 'php artisan make:auth';
+
+        $this->info("Executing $command");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        $this->line($process->getOutput());
+    }
+
+    protected function doValetLink()
+    {
+        if (! $this->tools['valet']) {
+            $this->warn('Cannot find valet on your system so a valet link was not created.');
+
+            return false;
+        }
+
+        $command = "valet link {$this->projectname}";
+        $this->info("Linking valet by executing '$command' in {$this->basepath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        $this->line($process->getOutput());
+        return true;
+    }
+
+    protected function doGit()
+    {
+        if ($this->option('message')) {
+            $this->commitmessage = $this->option('message');
+        }
+
+        if (! $this->tools['git']) {
+            $this->info("Unable to find 'git' on the system so I cannot initialize a git repo in '{$this->projectpath}'");
+
+            return false;
+        }
+
+        $process = new Process('dummy command');
```

Comment:
Is there any cost to making a new process each time?

Reply:
Fixed

Reply:
I'm not sure but this was the best way I could come up with. I was surprised to not find a nice straightforward way to build up multiple commands like this but I'm likely missing something.

I remain hopeful that time will improve the command execution part of the code, especially as Laravel Zero grows to help handle such things. The "dummy command" seems strange as it could instead be an empty string but I'll leave it in this PR :)

Reply:
What if we replaced it with this?

```php
        $commands = [
            'git init',
            'git add .',
            'git commit -m "' . str_replace('"', '\"', $this->commitmessage) . '"',
        ];

        foreach ($commands as $command) {
            $process = new Process($command);
            $process->setWorkingDirectory($this->projectpath);
            $process->run();
            $this->line($process->getOutput());
        }
```

or this?

```php
        $message = str_replace('"', '\"', $this->commitmessage);
        $process = new Process('git init && git add . && git commit -m "' . $message . '"');
        $process->setWorkingDirectory($this->projectpath);
        $process->run();
        $this->line($process->getOutput());
```